### PR TITLE
feat: added Faq section in overview

### LINKED ISSUE #190: cookbook page, with solutions to frequently asked questions
I am opening this as an issue rather than a PR to start, in order to gather feedback and additional content.

There are a number of questions that are asked frequently on stackoverflow, slack, github issues etc that come down to "how do I express this pattern in JSON Schema?"  Here are a few of them, and I'm sure we can come up with more:

- "evaluate this subschema but do not collect annotations" (not so common but it has come up a few times, and it is weird enough that it deserves a mention IMO):  
`{ not: { not: { ... subschema ... } } }`

- add descriptions to a bunch of enum values, e.g. for form generation:
```
  {
    anyOf: [
      {
        "description": "...",
        "const": "value0",
      },
      {
        "description": "...",
        "const": "value1",
      },
      ...
    ]
  }
```

- simulate OpenAPI's "discriminator":
```
  {
    allOf: [
      {
        if: {
          type: object,
          required: [ "foo" ],
          properties: { foo: { const: "value0" } }
        },
        then: {
          .. subschema when data has foo: value0
        }
      },
      {
        if: {
          type: object,
          required: [ "foo" ],
          properties: { foo: { const: "value1" } }
        },
        then: {
          .. subschema when data has foo: value1
        }
      },
      {
        if: {
          type: object,
          required: [ "bar" ],
          properties: { bar: { const: "value0" } }
        },
        then: {
          .. subschema when data has bar: value0
        }
      },
      ...
    ],
}
```

### PR DESCRIPTION
**Pull request by Akshaybagai52**
**What kind of change does this PR introduce?**

**Issue Number:** #190
-  Closes #190 

**Screenshots/videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

### CODE REVIEW

File: pages/frequently-asked-questions/index.page.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React, { useState, useEffect } from 'react'
```
Comment (benjagm) [inline pages/frequently-asked-questions/index.page.tsx:1]: * This page will go under the overview section and with a link in the sidebar.
pages/overview/faq

* Can we rename frequently-asked-questions by faq?

* This page is missing the sidebar. Please see [slug].page.tsx inside overview to understand how to show it.

* I'd love to provide a experience like to this page: https://graphql.org/faq/ With groups, collapsing expanding effect and each question will have a mardown heading to able to generate self-links so people can link the questions externaly.

**Review by benjagm — CHANGES_REQUESTED** [review]
This is a great start. Thanks a lot!! I left some comment but this is moving in the right direction. Congrats!!

**Comment by Akshaybagai52** [comment]
Thanks @benjagm for the review will implement these changes

File: data/faq.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
+[
```
[the remark this answers is not in the document]
Comment (Akshaybagai52) [reply data/faq.json:1]: I've completed this

File: pages/frequently-asked-questions/index.page.tsx (hunk 1 above)
Comment (Akshaybagai52) [reply pages/frequently-asked-questions/index.page.tsx]: This is also done

**Comment by Akshaybagai52** [comment]
Hey @benjagm can you review this 
Thanks

**Comment by benjagm** [comment]
Can you please change the target branch of this PR to : https://github.com/json-schema-org/website/tree/web-release-3

Reference docs: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request

**Comment by Akshaybagai52** [comment]
Hey @benjagm I've changed the base branch can you review this now

File: components/Sidebar.tsx [hunk 3]
```diff
@@ -69,6 +69,7 @@ const getDocsPath = [
   '/overview/sponsors',
   '/overview/similar-technologies',
   '/overview/code-of-conduct',
+  'overview/faq',
```
Comment (benjagm) [inline components/Sidebar.tsx:72]:
```suggestion
  '/overview/faq',
```

Comment (benjagm) [reply components/Sidebar.tsx:72]: Missing '/' character.

File: components/Accordion.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    <div>
+      {items.map((item, index) => (
+        <div
+          key={index}
+          className={`overflow-hidden transition-max-height border-t-2 ${
+            activeIndex === index ? 'max-h-96' : 'max-h-20'
+          }  ${index === items.length - 1 ? 'border-b-2' : ''}`}
+        >
+          <div
+            className='flex justify-between items-center p-4 cursor-pointer'
```
Comment (benjagm) [inline components/Accordion.tsx:29]: Can we use the icon '+' (plus) instead of the triangle?

File: components/Faq.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
```
Comment (benjagm) [inline components/Faq.tsx:1]: If accordion and faq are not reusable components I suggest to keep the functionality inside the faq tsx page.

File: components/Accordion.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+            onClick={() => handleToggle(index)}
+          >
+            <div className='text-[20px]'>{item.question}</div>
```
Comment (benjagm) [inline components/Accordion.tsx:32]: For questions to have their own externally accesible url well need to add a self link something like:

````
<div className='text-[20px]'><a href="#{question-id}">{item.question}</a></div>
````

**Review by benjagm — CHANGES_REQUESTED** [review]
This is almost done! Excellent job here @Akshaybagai52 !!

Just left the last changes but this is pretty close to be completed.

File: components/Sidebar.tsx (hunk 3 above)
Comment (Akshaybagai52) [reply components/Sidebar.tsx]: I've updated this

File: components/Accordion.tsx (hunk 4 above)
Comment (Akshaybagai52) [reply components/Accordion.tsx:30]: Updated this

File: components/Faq.tsx (hunk 5 above)
Comment (Akshaybagai52) [reply components/Faq.tsx:1]: I made it like that because in future it can become reusable components. If any other page needs this accordion and also we can add the question answer by passing category as props so it can be reused if we have some other categories of questions like we are doing with "general"

File: components/Accordion.tsx (hunk 6 above)
Comment (Akshaybagai52) [reply components/Accordion.tsx:33]: What I did in this is that on clicking of question the accordion expands and then show the answer of the quesiton but with this addition it will redirect to other link. 
So what I did in this is to add arrow in the end of answer on clicking of that arrow they can redirect to respected links. 
Let me know about your views on this

File: components/CarbonsAds.tsx [hunk 7]
```diff
@@ -115,6 +115,12 @@ CarbonAds.stylesheet = {
       margin-top: 4px;
       color: rgb(100 116 139);
     }
+
+    @media (max-width: 1023px) {
```
Comment (benjagm) [inline components/CarbonsAds.tsx:119]: Why pushing this change? please remove from the PR anything that is unrelated to this feature.

File: components/Layout.tsx [hunk 8]
```diff
@@ -329,128 +329,6 @@ const Footer = () => (
   </footer>
 );
 
-const OpenJS = () => (
```
Comment (benjagm) [inline components/Layout.tsx:332]: Why pushing this change? please remove from the PR anything that is unrelated to this feature.

File: pages/understanding-json-schema/basics.md [hunk 9]
```diff
@@ -139,8 +139,8 @@ you control, for example:
 { "$id": "http://yourdomain.com/schemas/myschema.json" }
```
Comment (benjagm) [inline pages/understanding-json-schema/basics.md:139]: Why pushing this change? please remove from the PR anything that is unrelated to this feature.

File: pages/understanding-json-schema/structuring.md [hunk 10]
```diff
@@ -8,7 +8,7 @@ section: docs
 * [Schema Identification](#schema-identification)
```
Comment (benjagm) [inline pages/understanding-json-schema/structuring.md:8]: Why pushing this change? please remove from the PR anything that is unrelated to this feature.

File: components/Accordion.tsx (hunk 6 above)
Comment (benjagm) [reply components/Accordion.tsx:33]: The accordion expansion effects works well but we need to also change the url from /faq to something like  /faq#xxxxxxxxxx. The blue arrow you added I think doesn't add any value. Just focus on adding the self link navigation when expanding the accordion.

**Review by benjagm — CHANGES_REQUESTED** [review]
We are almost there! Left some comments.

File: .github/workflows/greet_on_first_merge.yml [hunk 11]
```diff
@@ -0,0 +1,38 @@
+name: Greet on User First PR Merge
```
[the remark this answers is not in the document]
Comment (Akshaybagai52) [reply .github/workflows/greet_on_first_merge.yml:1]: actually these changes comes automatically when I change the base branch

**Comment by Akshaybagai52** [comment]
Hey @benjagm these changes comes automatically when I changed the base branch from main to web-release-3

File: .github/workflows/greet_on_first_merge.yml (hunk 11 above)
[the remark this answers is not in the document]
Comment (benjagm) [reply .github/workflows/greet_on_first_merge.yml:1]: That is fine but remove them from your PR. Just push your code

[the remark this answers is not in the document]
Comment (Akshaybagai52) [reply .github/workflows/greet_on_first_merge.yml:1]: Okay

File: components/Accordion.tsx (hunk 6 above)
Comment (Akshaybagai52) [reply components/Accordion.tsx:33]: Got it I'll update this

**Review by benjagm — CHANGES_REQUESTED** [review]
We are getting really closer. Congrats!!

This is looking really good!!!!

I found two issue. See the video for extended explanation:

1. Unexpected contraction of the page width. 
2. Links to specific questions doesn't work.

Video:

https://github.com/json-schema-org/website/assets/40007659/41daee13-08f4-4c86-b20b-31dd4fcead8c

**Comment by Akshaybagai52** [comment]
> We are getting really closer. Congrats!!
> 
> This is looking really good!!!!
> 
> I found two issue. See the video for extended explanation:
> 
> 1. Unexpected contraction of the page width.
> 2. Links to specific questions doesn't work.
> 
> Video with audio:
> 
>  video1444402921.mp4

Thanks for the detailed review I will fix this

**Comment by Akshaybagai52** [comment]
Hey @benjagm when I reset branch then PR got closed 
The issue is resolved now you can check in this PR
https://github.com/json-schema-org/website/pull/534
